network-api/networkapi/wagtailpages/pagemodels/customblocks/text_only_teaser.py

```diff
@@ -1,13 +1,30 @@
+from django.forms.utils import ErrorList
 from wagtail.core import blocks
+from wagtail.core.blocks.struct_block import StructBlockValidationError
 
 
 class TextOnlyTeaserCard(blocks.StructBlock):
 
     heading = blocks.CharBlock(help_text="Heading for the card.")
     heading_link = blocks.PageChooserBlock(required=False, help_text="Page that the header should link out to.")
+    link_url = blocks.CharBlock(
+        required=False,
+        help_text="Optional URL that the header should link out to.",
+    )
     meta_data = blocks.CharBlock(max_length=500)
     body = blocks.TextBlock(help_text="Body text of the card.", required=False, max_length=200)
 
+    def clean(self, value):
+        result = super().clean(value)
+        errors = {}
+
+        if value["heading_link"] and value["link_url"]:
+            errors["heading_link"] = ErrorList(["Please choose between a heading link OR a URL value."])
+        if errors:
+            raise StructBlockValidationError(errors)
```

Comment:
Can you add some context why are not raising the `StructBlockValidationError` directly when we detect the issue? 


```suggestion
        if value["heading_link"] and value["link_url"]:
            raise StructBlockValidationError(
              {"heading_link": ErrorList(["Please choose between a heading link OR a URL value."])}
            )
```

Reply:
This is just preference I think, it allows for other errors to be added into the `errors` dict. I'll rework this to be more specific though

Reply:
Fixed in bdf625d3490912c61bdc08dee10fa43f3d077e9b